6.analyze_features/scripts/8c.custom_aggregated_PCA_calculate_mahalanobis_distance.py

```diff
@@ -306,54 +308,110 @@
 )
 
 
+# In[13]:
+
+
+# anova test the mahalanobis distance between the genotypes
+anova_result_across_genotypes = anova(
+    high_severity_mahalanobis_distances,
+    low_severity_mahalanobis_distances,
+    wt_mahalanobis_distances,
+)
+
+print(
+    f"The p-value for the ANOVA across the genotypes is {anova_result_across_genotypes.pvalue}"
+)
+
+# tukeys HSD test for the mahalanobis distance between the genotypes
+tukeys_result_across_genotypes = pairwise_tukeyhsd(
+    np.concatenate(
+        [
+            high_severity_mahalanobis_distances,
+            low_severity_mahalanobis_distances,
+            wt_mahalanobis_distances,
+        ]
+    ),
+    np.concatenate(
+        [
+            ["High Severity"] * len(high_severity_mahalanobis_distances),
+            ["Low Severity"] * len(low_severity_mahalanobis_distances),
+            ["Wild Type"] * len(wt_mahalanobis_distances),
+        ]
+    ),
+)
+# get the tukeys HSD results as a dataframe
+tukeys_result_across_genotypes_df = pd.DataFrame(
+    data=tukeys_result_across_genotypes._results_table.data[1:],
+    columns=tukeys_result_across_genotypes._results_table.data[0],
+)
+tukeys_result_across_genotypes_df
+
+
 # ### write the Mahanobis distance stats to a file
 
-# In[13]:
+# In[14]:
 
 
 # set the output dir
-mahalanobis_output_dir = pathlib.Path(
-    "../results/custom_aggregation_results/"
-).resolve()
+mahalanobis_output_dir = pathlib.Path("../results/mean_aggregation_results/").resolve()
 # make the dir if it does not exist
 mahalanobis_output_dir.mkdir(parents=True, exist_ok=True)
 
 # define the output file path
 mahalanobis_output_file_path = pathlib.Path(
-    mahalanobis_output_dir / "custom_aggregation_mahalanobis_distance_results.csv"
+    mahalanobis_output_dir / "mean_aggregation_mahalanobis_distance_results.csv"
```

Comment:
Shouldn't this output be "custom" not "mean" since this notebook references the custom aggregation?